library/net_sockets.c

```diff
@@ -270,13 +270,18 @@ static int net_would_block( const mbedtls_net_context *ctx )
  */
 static int net_would_block( const mbedtls_net_context *ctx )
 {
+    int err = errno;
+    
     /*
      * Never return 'WOULD BLOCK' on a non-blocking socket
      */
     if( ( fcntl( ctx->fd, F_GETFL ) & O_NONBLOCK ) != O_NONBLOCK )
```

Comment:
I like @gilles-peskine-arm comment, but I would also like to say that if the file descriptor is invalid, the code would have failed much earlier anyways. So, I agree this is not an urgent problem, just nice-to-have.